cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -478,6 +491,54 @@ impl OpenshiftSecondaryMetadataParserPlugin {
         Ok(())
     }
 
+    async fn process_conditional_edges(
+        &self,
+        graph: &mut cincinnati::Graph,
+        data_dir: &PathBuf,
+    ) -> Fallible<()> {
+        let blocked_edges_dir = data_dir.join(BLOCKED_EDGES_DIR);
+        let conditional_edges: Vec<graph_data_model::ConditionalEdgeYaml> =
+            deserialize_directory_files(
+                &blocked_edges_dir,
+                regex::Regex::new("ya+ml")?,
+                &self.settings.disallowed_errors,
+            )
+            .await
+            .context(format!(
+                "looking for conditional edges in blocked edges {:?}",
+                blocked_edges_dir
+            ))?;
+
+        debug!(
+            "Found {} valid conditional edges declarations.",
+            conditional_edges.len()
+        );
+
+        conditional_edges
+            .into_iter()
+            .try_for_each(|cey| -> Fallible<()> {
+                let ce: ConditionalEdge = ConditionalEdge {
+                    edge_regex: ConditionalUpdateEdge {
+                        from: cey.from.to_string(),
+                        to: cey.to.to_string(),
+                    },
+                    edges: vec![],
+                    // Some edges could be blocked for multiple risks, here we are parsing each yaml
+                    // file separately. TODO: remove edge-risk duplication
```

Comment:
You need to do at least some condensing here.  Constraints:

* Each YAML file contains a single risk, mentioning a single target release and a regexp of source releases.
* To build the conditional edge for a single target, you'll need a single `conditionalEdges` entry with that target release in `edges[].to`, including the complete set of relevant `risks`.  Collapsing all the risks for a single edge into one `conditionalEdges` entry is required, not optional.
* The risk-set may be shared between multiple edges (e.g. A -> C and B -> C could have the same risk sets).  Collapsing multiple edges into the same `conditionalEdges` entry is an optional, space-saving mechanism.

Assuming we defer multi-edge collapse, you'll still need to collapse multiple risks for each single edge.  You can do a bit of that here, by collapsing to a single `conditionalEdges` entry for each `(from, to)` tuple.  But because `from` is a regular expression here, and different regular expressions might match the same actual-`from` release, you'll still need a separate multi-risk collapse later on, after the regular expressions have been validated.

Reply:
handled this use case in `edge-add-remove` plugin